netlogo-gui/src/main/properties/ResourcePathEditor.scala

```diff
@@ -0,0 +1,89 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.properties
+
+import java.awt.{ Color, Component, FileDialog => JFileDialog }
+import java.awt.event.{ ActionEvent, MouseAdapter, MouseEvent }
+import javax.swing.{ AbstractAction, BorderFactory, JButton, JLabel, JPanel, JPopupMenu }
+
+import org.nlogo.api.ExternalResourceManager
+import org.nlogo.awt.UserCancelException
+import org.nlogo.core.{ ExternalResource, I18N }
+import org.nlogo.swing.{ FileDialog, OptionDialog }
+
+abstract class ResourcePathEditor(accessor: PropertyAccessor[ExternalResource.Location], useTooltip: Boolean,
+                                  parent: Component, resourceManager: ExternalResourceManager)
+    extends PropertyEditor(accessor, useTooltip) {
+
+  private class ResourceMenu extends JPanel {
+
+    private val label = new JLabel(emptyName)
+
+    private var resourceOpt: Option[ExternalResource.Location] = None
+
+    setBorder(BorderFactory.createLineBorder(Color.BLACK))
+
+    add(label)
+    add(new JLabel("v")) // placeholder until gui redesign is merged
```

Comment:
:eyes:

The comment should go, and/or the functionality should be changed to something that we're happy (enough) with.

Reply:
There is the perfect class for this in the GUI PR, and I would really like to use it here. Is it ok to make a note of that change elsewhere and remove the comment for now?